sync/syncer.go

```diff
@@ -416,9 +423,9 @@ func (s *Syncer) syncTxs(txids []types.TransactionId) ([]*types.AddressableSigne
 	unprocessedTxs := make(map[types.TransactionId]*types.AddressableSignedTransaction)
 	missing := make([]types.TransactionId, 0)
 	for _, t := range txids {
-		if tx := s.txpool.Get(t); tx != nil {
+		if tx, err := s.txpool.Get(t); err == nil {
 			s.Info("found tx, %v in tx pool", hex.EncodeToString(t[:]))
-			unprocessedTxs[t] = tx.(*types.AddressableSignedTransaction)
+			unprocessedTxs[t] = &tx
```

Comment:
why do you need the address of tx ? (i know `unprocessedTxs` is id to pointer but maybe you dont need it).

Reply:
to keep consistency and not change or refactor any outside dependencies I've decided to keep it this way.